tensorboard/uploader/flags_parser.py

```diff
@@ -201,3 +203,11 @@ def define_flags(parser):
     auth_revoke.set_defaults(
         **{AUTH_SUBCOMMAND_FLAG: AUTH_SUBCOMMAND_KEY_REVOKE}
     )
+
+
+def fix_flags(flags):
```

Comment:
Instead of adding a `fix_flags` mechanism, can we just use the `type`
kwarg to `parser.add_argument` to control the parsing ourselves? Like:

```python
    upload.add_argument(
        "--plugins",
        type=lambda s: s.split(","),
        default=[],
        help="...",
    )
```

and then you’re just done, I think?

If the parsing is nontrivial and needs to be done in multiple places,
you can add a helper function, like `_parse_samples_per_plugin` from
`backend/application.py`. But I’m not sure that adding a whole new axis
to the subcommand layer is necessary.

Reply:
Yes, this is trivial if all we want is comma-separated lists.

Reply:
Even if we wanted to support both comma-separated and tokenized lists,
it would be easy to accomplish with a simple custom action:

```python
import argparse


class ExtendFromAction(argparse.Action):
    def __call__(self, parser, namespace, values, option_string=None):
        items = getattr(namespace, self.dest, None) or []
        items.extend(x for xs in values for x in xs)
        setattr(namespace, self.dest, items)


parser = argparse.ArgumentParser()
parser.add_argument(
    "--plugins", nargs="*", action=ExtendFromAction, type=lambda x: x.split(",")
)

print(parser.parse_args(["--plugins", "a,b", "c,d", "--plugins", "e,f"]))
# prints: Namespace(plugins=['a', 'b', 'c', 'd', 'e', 'f'])
```

Or, we could just use `nargs="*", type=lambda x: x.split(",")` and
flatten with a helper function in the application code.

The `fix_flags` used by the plugin system is mainly used to signal
errors due to _interactions_ of flags: “if inspect-mode is set but not
exactly one of a logdir or event file is given, then die”. As long as
all we’re doing is controlling the parsing of a _single_ flag, I’d
prefer to do so with `type=` or `action=` or similar rather than adding
a new mechanism.